frontend/lib/uast/post-parse-checks.cpp

```diff
@@ -709,6 +711,59 @@ void Visitor::checkDmappedKeyword(const OpCall* node) {
            " instead please use factory functions when available");
 }
 
+static int getOpCallPrecedence(const OpCall* node) {
+  return opToPrecedence(node->op(), /* unary */ false, /* postfix */ false);
+}
+
+static void collectEqualPrecedenceOpsWithoutParens(Context* context,
+                                                   const OpCall* node,
+                                                   int prec,
+                                                   std::vector<const OpCall*>& ops,
+                                                   std::vector<const AstNode*>& operands) {
+  auto check = [context, prec, &ops, &operands](const AstNode* child) {
+    if (auto childOp = child->toOpCall()) {
+      if (childOp->numActuals() == 2 && getOpCallPrecedence(childOp) == prec) {
+        // The child only counts as a 'problem' if it's not parenthesized.
+        if (parsing::locateExprParenthOpenWithAst(context, childOp).line() == -1) {
+          collectEqualPrecedenceOpsWithoutParens(context, childOp, prec, ops, operands);
+          return;
+        }
+      }
+    }
+
+    operands.push_back(child);
+  };
+
+  check(node->actual(0));
+  ops.push_back(node);
+  check(node->actual(1));
+}
+
+void Visitor::checkNonAssociativeComparisons(const OpCall* node) {
+  if (node->numActuals() != 2) return;
+
+  auto lessThanPrec = opToPrecedence(USTR("<"), /* unary */ false, /* postfix */ false);
```

Comment:
Consider just `const bool unary = false;` and `const bool postfix = false;`.

Reply:
I've combined this into 

```
static int binOpPrecedence(UniqueString ustr) {
  bool unary = false;
  bool postfix = false;
  return opToPrecedence(ustr, unary, postfix);
}
```

Which makes it no longer a one-liner, and saves some repetition.